python/dgl/transform.py

```diff
@@ -2097,15 +2103,36 @@ def to_block(g, dst_nodes=None, include_dst_in_src=True):
         if g._graph.ctx != d.ctx:
             raise ValueError('g and dst_nodes need to have the same context.')
 
-    new_graph_index, src_nodes_nd, induced_edges_nd = _CAPI_DGLToBlock(
-        g._graph, dst_node_ids_nd, include_dst_in_src)
+    src_node_ids = None
+    src_node_ids_nd = None
+    if src_nodes is not None and not isinstance(src_nodes, Mapping):
+        # src_nodes is a Tensor, check if the g has only one type.
+        if len(g.ntypes) > 1:
+            raise DGLError(
+                'Graph has more than one node type; please specify a dict for src_nodes.')
+        src_nodes = {g.ntypes[0]: src_nodes}
+        src_node_ids = [
+            utils.toindex(src_nodes.get(ntype, []), g._idtype_str).tousertensor(
+                ctx=F.to_backend_ctx(g._graph.ctx))
```

Comment:
I feel you can just use `F.copy_to(F.tensor(...), F.to_backend_ctx(...))`.  If the input tensor is already on the input device it does not actually perform a copy.

Reply:
Changed to `F.copy_to()` in 33b5d14.